### ISSUE TITLE
[java] InefficientEmptyStringCheck false negatives

### ISSUE THREAD

**Issue by maikelsteneker** [pmd/pmd#4013]
**Affects PMD Version:** 6.46.0

**Rule:** [InefficientEmptyStringCheck](https://pmd.github.io/latest/pmd_rules_java_performance.html#inefficientemptystringcheck)

**Description:** When upgrading from PMD 6.33.0 to the latest version (6.46.0), we came across multiple false negatives that seem to have been introduced by PR #3344.

**Code Sample demonstrating the issue:**

```
public class MyClass {
  public MyClass(String var) {
    if( var.trim().length() > 0 ) { // false negative
      config.set( var, true );
      var = token;
    }
    if( var != null && var.trim().length() > 0 ) { // false negative
      config.set( var, true );
    }
  }
}
```

**Expected outcome:** Violations on the guards of the two if statements above.

**Running PMD through:** *CLI*

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7.0.0
